src/Link.php

```diff
@@ -539,50 +396,122 @@ public static function generateLinkContents($link, CommonDBTM $item, bool $safe_
             }
         }
 
-        $links = [];
-        if (count($ipmac) > 0) {
-            foreach ($ipmac as $key => $val) {
-                $tmplink = $link;
-                $disp    = 1;
-                if (strstr($link, "[IP]")) {
-                    if (empty($val['ip'])) {
-                        $disp = 0;
-                    } else {
-                        $tmplink = str_replace("[IP]", $val['ip'], $tmplink);
-                    }
-                }
-                if (strstr($link, "[MAC]")) {
-                    if (empty($val['mac'])) {
-                        $disp = 0;
-                    } else {
-                        $tmplink = str_replace("[MAC]", $val['mac'], $tmplink);
-                    }
-                }
+        return $ipmac;
+    }
 
-                if ($disp) {
-                    if ($safe_url) {
-                        $tmplink = URL::sanitizeURL($tmplink) ?: '#';
-                    }
-                    $links[$key] = $tmplink;
-                }
+    /**
+     * Generate link(s).
+     *
+     * @param string        $link           original string content
+     * @param CommonDBTM    $item           item used to make replacements
+     * @param bool          $safe_url       indicates whether URL should be sanitized or not
+     * @param array         $custom_vars    custom variables that will be passed to link template renderer
+     *
+     * @return array of link contents (may have several when item have several IP / MAC cases)
+     */
+    public static function generateLinkContents($link, CommonDBTM $item, bool $safe_url = true, array $custom_vars = [])
```

Comment:
The HTML special chars (`>`, `<` and `&`) contained in templates should be encoded, to ensure that they will not permit to generate HTML. For instance, if I put `<script>alert(1)</script>` in a link template, it will be be executed.

In GLPI 10.0, these chars were encoded by the autosanitization logic, but it has been removed.

A test case should be added for that.

Reply:
The link itself is still safe it seems (replaced with #). However, I did find that we need to address sanitization in the search results display. That issue seems global and not related to this PR.

Reply:
Fixed in last commit. The issue was with a link that contains a file template.